File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Seeds.java
package com.logistics.Components.Product;

public class Seeds extends Flora {

    private int expirationDays;

    public Seeds() {
        super();
        expirationDays = 0;
    }

    public Seeds(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, boolean isCultivated, int expirationDays) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, isCultivated);
        this.expirationDays = expirationDays;
    }

    // Copy Constructor
    public Seeds(Seeds other) {
        super(other);
        this.expirationDays = other.expirationDays;
    }

    // Clone method
    @Override
    public Seeds clone() {
        return new Seeds(this);
    }

    // Getters
    public int getExpirationDays() { return expirationDays; }

    // Setters
    public void setExpirationDays(int expirationDays) { this.expirationDays = expirationDays; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nExpiration Days: " + expirationDays;
    }
}

File: Task 1/C++/Components/Product/Seeds.cpp
#include "Seeds.h"
Seeds::Seeds() : Flora(){
    expirationDays = 0;
}

Seeds::Seeds(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, bool isCultivated, int expirationDays) 
        : Flora(name, cost, supplier, discount_amount, packing_units, cost_per_unit, isCultivated){
    this->expirationDays = expirationDays;
}

Seeds::Seeds(const Seeds& other) : Flora(other) {
    this->expirationDays = other.expirationDays;
}

std::string Seeds::display_product_info(){
    return Flora::display_product_info()+
            "\nExpiration Days: " + std::to_string(expirationDays);
}